cpp/lib/src/ZoteroHarvesterConfig.cc

```diff
@@ -143,49 +180,60 @@ GlobalParams::GlobalParams(const IniFile::Section &config_section) {
     }
-}
 
+    // Warnings for unknown entries
+    for (const auto &entry : config_section) {
+        if (entry.name_.empty())
+            continue;
 
-const std::map<GlobalParams::IniKey, std::string> GlobalParams::KEY_TO_STRING_MAP {
-    { ENHANCEMENT_MAPS_DIRECTORY,                 "enhancement_maps_directory" },
-    { GROUP_NAMES,                                "groups" },
-    { STRPTIME_FORMAT_STRING,                     "common_strptime_format" },
-    { SKIP_ONLINE_FIRST_ARTICLES_UNCONDITIONALLY, "skip_online_first_articles_unconditionally" },
-    { DOWNLOAD_DELAY_DEFAULT,                     "default_download_delay_time" },
-    { DOWNLOAD_DELAY_MAX,                         "max_download_delay_time" },
-    { REVIEW_REGEX,                               "zotero_review_regex" },
-    { RSS_HARVEST_INTERVAL,                       "journal_rss_harvest_interval" },
-    { RSS_FORCE_PROCESS_FEEDS_WITH_NO_PUB_DATES,  "force_process_feeds_with_no_pub_dates" },
-    { TIMEOUT_CRAWL_OPERATION,                    "timeout_crawl_operation" },
-    { TIMEOUT_DOWNLOAD_REQUEST,                   "timeout_download_request" },
-};
-
+        bool valid(false);
+        for (const auto &allowed_value : GlobalParams::KEY_TO_STRING_MAP) {
+            if (entry.name_ == allowed_value.second) {
+                valid = true;
+                break;
+            }
+        }
 
-std::string GlobalParams::GetIniKeyString(const IniKey ini_key) {
-    const auto key_and_string(KEY_TO_STRING_MAP.find(ini_key));
-    if (key_and_string == KEY_TO_STRING_MAP.end())
-        LOG_ERROR("invalid GlobalParams INI key '" + std::to_string(ini_key) + "'");
-    return key_and_string->second;
+        if (not valid and not ZoteroMetadataParams::IsValidIniEntry(entry) and not MarcMetadataParams::IsValidIniEntry(entry))
+            LOG_WARNING("Invalid ini entry \"" + entry.name_ + "\" in global section");
+    }
 }
 
 
+const std::map<GroupParams::IniKey, std::string> GroupParams::KEY_TO_STRING_MAP {
+    { USER_AGENT,                       "user_agent" },
+    { ISIL,                             "isil" },
+    { OUTPUT_FOLDER,                    "output_folder" },
+    { AUTHOR_SWB_LOOKUP_URL,            "author_swb_lookup_url" },
+    { AUTHOR_LOBID_LOOKUP_QUERY_PARAMS, "author_lobid_lookup_query_params" },
+};
+
+
 GroupParams::GroupParams(const IniFile::Section &group_section) {
     name_ = group_section.getSectionName();
     user_agent_ = group_section.getString(GetIniKeyString(USER_AGENT));
     isil_ = group_section.getString(GetIniKeyString(ISIL));
     output_folder_ = group_section.getString(GetIniKeyString(OUTPUT_FOLDER));
     author_swb_lookup_url_ = group_section.getString(GetIniKeyString(AUTHOR_SWB_LOOKUP_URL));
     author_lobid_lookup_query_params_ = group_section.getString(GetIniKeyString(AUTHOR_LOBID_LOOKUP_QUERY_PARAMS), "");
-}
 
+    // Warnings for unknown entries
+    for (const auto &entry : group_section) {
```

Comment:
This looks exactly like some other new code further up.  Please encapsulate both in a new function.

Reply:
There are 3 validation mechanisms looking similar but they are not exactly the same (additional functions are called for validation and the error messages differ as well)

Reply:
i see what i can do

Reply:
I could only achieve that with a template function. How do you define a generic parameter for a map of any enum to a std::string?

Reply:
Yes, but it is a trivial template.